### PR TITLE
[FIXUP] cmake: Improve interface libraries code

### PR DESCRIPTION
**Pull request by hebasto**
We use interface libraries to encapsulate global / common / group build options. However, the current `cmake-staging` branch has a few issues:
1. An interface library name should be more expressive for better reading and reasoning about the code (pointed by **theuni** and **TheCharlatan** during the recent offline meeting).
2. The `core` library is referenced in the "cmake: Check system symbols" commit, which comes before introducing it in the "cmake: Add platform-specific definitions and properties" one.
3. The `get_target_interface` helper function shouldn't belong to the `ProcessConfigurations` module.

This PR fixes all issues mentioned above.

Unfortunately, I failed to put this PR changes into the fixup / autosquash routine. So the plan is to undraft it just after the next sync to the master branch, then merge this PR and adjust the commit history in the following one.

### CODE REVIEW

**Comment by hebasto** [comment]
cc @theuni @TheCharlatan

**Comment by theuni** [comment]
Thanks for the renames. LGTM.

File: CMakeLists.txt [hunk 1]
```diff
@@ -244,60 +244,60 @@ endif()
...
 # gccbug_90348 test case (only reproduces on GCC 11 and earlier) and
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111843. To work around that, set
 # -fstack-reuse=none for all gcc builds. (Only gcc understands this flag).
-try_append_cxx_flags("-fstack-reuse=none" TARGET core)
+try_append_cxx_flags("-fstack-reuse=none" TARGET core_interface)
 
 if(HARDENING)
-  add_library(hardening INTERFACE)
+  add_library(hardening_interface INTERFACE)
+  target_link_libraries(core_interface INTERFACE hardening_interface)
```
Comment (TheCharlatan) [inline CMakeLists.txt:257]: Why is this moved up here?

Comment (hebasto) [reply CMakeLists.txt:257]: To make both lines visually connected. The benefit of interface libraries is that the `target_link_libraries` command might be placed anywhere after the `add_library` one.

**Review by TheCharlatan — APPROVED** [review]
LGTM ACK 1e6c90fffcaeb6bbaceae564db62a2c7c436abfe

**Comment by hebasto** [comment]
> So the plan is to undraft it just after the next sync to the master branch, then merge this PR and adjust the commit history in the following one.

Please see https://github.com/hebasto/bitcoin/pull/76. That PR is expected to got, not this one.

There must be zero diff between them.

**Comment by hebasto** [comment]
Closing. See https://github.com/hebasto/bitcoin/pull/76#issuecomment-1893737052.
